### ISSUE TITLE
Bug: analysis suggestion illegal (g1f3)

### ISSUE THREAD

**Issue by ddugovic** [lichess-org/lichobile#1101]
I'm submitting a bug report.

**Current behavior:**

After game https://lichess.org/3mAwseEg/black was decided by timeout, engine analysis suggested an illegal move:
![image](https://user-images.githubusercontent.com/1834123/71472577-31f76d80-2799-11ea-8c54-420175e6b7e4.png)

**Expected behavior:**

Engine analysis should display correctly or not display at all

**Steps to reproduce:**

> It took me playing almost ten thousand games though to find this bug...
> I think from the information displayed it is easy to find this game. It ended by white exceeding time, then I started analysis and this happened. Stopped me in my tracks. It is white to move and Lichess’ advice to white is, how shall I say it, hard to follow...

**Environment information**

> iOS13.1.3, iPhone 11 Pro Max, 512 GB
> My Lichess App download is from 26 September 2019

**Other information:**

> I've seen this before elsewhere. I think it's even been posted somewhere in the discord before.
> Not for lichobile, but desktop.
> Usually disappears after running stockfish for a second.

**Comment by woldaker** [comment]
This happens to me _very_ often.  It's difficult to ascertain a set of conditions to reproduce this, as it seems to happen in a multitude of situations.  As far as this bug "usually disappearing after running stockfish for a second", I am not sure if I know what "running stockfish for a second" means.  I have my eval settings set to:
{run local evaluation=true, best move arrow=true, show/hide comments=true, infinite analysis=false, multiple lines=1, CPUs=1}

However, here are a few things I think I _can_ say about the bug:

1) It appears to be "stuck" at a certain move in the game.  I think a more accurate title for this bug report would be "analysis suggestions stuck at specific move".  That g1->f3 move in the example picture was, I'm pretty sure anyway, simply a previous move suggestion.

2) Requesting a computer evaluation (what I thought, perhaps, was meant by "running stockfish for a second") does not fix the issue.

3) Closing and reopening the analysis board does not fix the issue.

4) Closing and reopening the game in question, then reopening the analysis board does not fix the issue.

5) Even closing and reopening the Lichess app, finding that game in your games list, then reopening the analysis board does not fix the issue.

6) I have seen the +/- positional eval rating become stuck at that previous move as well, and I have also seen that return to the current move after a certain number of actions were taken e.g. closing the analysis board, game, app, or any/all of them.  I just don't know exactly what those actions are.
